crates/libs/version/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+#![cfg_attr(not(test), no_std)]
+
+mod bindings;
+use bindings::*;
+
+/// Operating system version information.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Version {
+    /// The major version number of the operating system.
+    pub major: u32,
+
+    /// The minor version number of the operating system.
+    pub minor: u32,
+
+    /// The major version number of the latest service pack installed on the system.
+    pub pack: u32,
+
+    /// The build number of the operating system.
+    pub build: u32,
+}
+
+impl Version {
+    /// Creates a new `Version` with the given values.
+    pub const fn new(major: u32, minor: u32, pack: u32, build: u32) -> Self {
+        Self {
+            major,
+            minor,
+            pack,
+            build,
+        }
+    }
+
+    /// Gets the version information of the currently running operating system.
+    #[cfg(not(test))]
+    pub fn current() -> Self {
+        unsafe {
+            let mut info = OSVERSIONINFOEXW {
+                dwOSVersionInfoSize: core::mem::size_of::<OSVERSIONINFOEXW>() as u32,
+                ..core::mem::zeroed()
+            };
+
+            RtlGetVersion(&mut info as *mut _ as *mut _);
+
+            Version {
+                major: info.dwMajorVersion,
+                minor: info.dwMinorVersion,
+                pack: info.wServicePackMajor as u32,
+                build: info.dwBuildNumber,
+            }
+        }
+    }
+
+    /// Hook used for testing `ge`.
+    #[cfg(test)]
+    fn current() -> Self {
+        test::test_current()
+    }
+}
+
+/// Determines if the currently running operating system is a Windows Server release.
+pub fn is_server() -> bool {
```

Comment:
It feels like this should be a method so you can do `current().is_server()`. However, after thinking about it I'm not sure the extra complexity would be worth it for a single function. We'd need another struct, say `OsInfo`, with a method for getting the version.

Reply:
It's a bit of an odd pairing but I can't think of something better than a free function without turning this into a bigger "info" crate.

Reply:
Yeah, I talked myself round to that point of view.